### PR TITLE
Added an integration test for lifecycle state transitions

### PR DESCRIPTION
**Pull request by yuchaoran2011**

### CODE REVIEW

File: test/e2e/lifecycle_test.go [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	"container/list"
+	"github.com/GoogleCloudPlatform/spark-on-k8s-operator/pkg/apis/sparkoperator.k8s.io/v1beta1"
```
Comment (liyinan926) [inline test/e2e/lifecycle_test.go]: Import order seems off.

File: test/e2e/lifecycle_test.go [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+	"github.com/stretchr/testify/assert"
+
+	"k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	appFramework "github.com/GoogleCloudPlatform/spark-on-k8s-operator/test/e2e/framework"
+)
+
+func TestLifeCycleManagement(t *testing.T) {
+	STATES := [9]string{
```
Comment (liyinan926) [inline test/e2e/lifecycle_test.go]: You can define this as a package level var.

Comment (yuchaoran2011) [reply test/e2e/lifecycle_test.go]: Done.

**Comment by liyinan926** [comment]
Please run `go fmt`.

**Pull request merged**